packages/flutter/lib/src/widgets/stretch_effect.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'image_filter.dart';
+
+/// A widget that replicates the native Android stretch overscroll effect.
+///
+/// This widget is used in the [StretchingOverscrollIndicator] widget and creates
+/// a stretch visual feedback when the user overscrolls at the edges.
+///
+/// Only supported when using the Impeller rendering engine.
+class StretchEffect extends StatefulWidget {
```

Comment:
Can you move this class to `overscroll_indicator` and make it private? I don't think we need this class for other public usages.